src/App.tsx

```diff
@@ -13,10 +13,34 @@ function App() {
   const [newTodo, setNewTodo] = useState('')
   const [dueDate, setDueDate] = useState<string>(new Date().toISOString().split('T')[0])
 
+  const loadTodosFromLocalStorage = (): Todo[] => {
+    const storedTodos = localStorage.getItem('todos')
+    if (storedTodos) {
+      return JSON.parse(storedTodos).map((todo: Todo) => ({
+        ...todo,
+        dueDate: new Date(todo.dueDate)
+      }))
+    }
+    return []
+  }
+
+  const saveTodosToLocalStorage = (todos: Todo[]) => {
+    localStorage.setItem('todos', JSON.stringify(todos))
+  }
+
+  useEffect(() => {
+    setTodos(loadTodosFromLocalStorage())
+  }, [])
+
+  useEffect(() => {
```

Comment:
I'm testing this and I can see that localStorage is being set when I add a new to do, but then when I refresh, localStorage is cleared - I think it is because 'saveTodosToLocalStorage' is firing on the initial load when todos is empty.